combat-trainer.lic

```diff
@@ -854,7 +861,8 @@ class LootProcess
   end
 
   def dissected?(mob_noun, game_state)
-    case bput("dissect #{mob_noun}","You'll gain no insights from this attempt", 'You succeed in dissecting the corpse','What exactly are you trying to dissect',"You'll learn nothing further",'While likely a fascinating study','You cannot dissect','would probably object',"should be left alone.","That's going to be hard","That'd be a waste of time.",'A skinned creature is worthless','You do not yet possess the knowledge', 'This ritual may only be performed on a corpse', /You learn something/i, 'A failed or completed ritual has rendered')
+    arrange_mob(DRRoom.dead_npcs.first, game_state) if @always_arrange
```

Comment:
The new instance var storing state and where all it gets read/written is hard for me to follow.

What do you think about moving all of that into `check_skinning`?

I'm picturing something like this:

```ruby
def check_skinning(mob_noun, game_state)
    already_arranged = false
    if @dissect && game_state.dissectable?(mob_noun)
      should_dissect = (@dissect_priority == 'First Aid' && DRSkill.getxp('First Aid') <= 32) ||
                                  (@dissect_priority == 'Skinning' && DRSkill.getxp('Skinning') >= 32) ||
                                  (DRSkill.getxp('First Aid') < DRSkill.getxp('Skinning')) ||
                                  (!game_state.skinnable?(mob_noun) && DRSkill.getxp('First Aid') <= 32) 
        if should_dissect
            if @always_arrange
                already_arranged = true
                arrange_mob(DRRoom.dead_npcs.first, game_state) 
            end
            return if dissected?(mob_noun, game_state)
        end
    end

    ... code ...

    arrange_mob(mob_noun, game_state) unless already_arranged
    already_arranged = false
```

Reply:
On second thought, I don't think that final `already_arranged = false` is even needed

Reply:
The reason for the instance var is that my next PR is to split the prioritization logic  out of `check_skinning` I've been working with Gildaren and Seped to setup a `should_skin_or_dissect` routine. I'm happy to put it as a local var in `check_skinning` for this PR, but then it'll change again.

Reply:
> I think the final `already_arranged = false` can be dropped in that approach?

It can be dropped in any case, Because the corpse can't exist beyond this point. It was either skinned or dissect, and we only come back to this point again in a new corpse, at which point `@already_arranged` is off again from `dispose_body`

Removed it.